Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -3873,7 +3873,7 @@ Vector<RenderLayer*> RenderLayer::topLayerRenderLayers(const RenderView& renderV
     Vector<RenderLayer*> layers;
     for (auto& element : renderView.document().topLayerElements()) {
         auto* renderer = element->renderer();
-        if (!renderer)
+        if (!renderer || renderer->isSkippedContent())
```

Comment:
This is not the right way to fix this. The correct approach is to have RenderTreeUpdater just not generate boxes in this case.

Reply:
We think it is the case because the spec mentions:

> If an element has an ancestor with [content-visibility: hidden](https://w3c.github.io/csswg-drafts/css-contain/#propdef-content-visibility), and it is placed in the [top layer](https://fullscreen.spec.whatwg.org/#top-layer), it does not generate any boxes, as if it were [display: none](https://w3c.github.io/csswg-drafts/css-display-3/#propdef-display).

Reply:
A very good point, I missed that completely! It makes the patch much smaller too.